sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -189,6 +189,7 @@ object AnsiTypeCoercion extends TypeCoercionBase {
         }
 
       case (DateType, TimestampType) => Some(TimestampType)
+      case (DateType, AnyTimestampType) => Some(TimestampType)
```

Comment:
shall we use `AnyTimestampType.defaultConcreteType`?

Reply:
The default concrete type is TimestampWithoutTZType. The current code is not ready for that yet. 
There will be a new flag for the default timestamp type in SQL. Let's revisit this later.

Reply:
I just change the default concrete type of AnyTimestampType to TimestampType, which makes more sense.